Add catches

mdhorn opened the pull request:
Changes proposed in this pull request:
- Addresses segmentation fault reported in Issue #145
- Improved code to avoid additional double free
- Improved error messages for users
- Exception handling to catch throws from included libraries

mdhorn committed:
Prevent double freeing of error data

Need to set the error point to NULL to ensure we do not
attempt to free a second time in the destructor.

This would occur if we were able to create a valid
rs_context and then later encounter a realsense runtime
error like failing to get an extrinsic.

mdhorn committed:
Prevent possible double free of rs_context

Make sure the rs_context is not freed twice; a second
time in the destructor.

mdhorn committed:
Remove unneeded exit() call

The code should be allow ros::shutdown() to properly
stop the node and call the destructors.

mdhorn committed:
Clearer error message for uvcmodule not loaded

Display a 'no camera detected' message when the uvcvideo module
is not being loaded. This is generally due to no camera every
being connected since last boot.

mdhorn committed:
Improve error messages

In general, attribute intrinsic failures to firmware, and
attribute eextrinsics failures to calibration issues.

mdhorn committed:
Added exception handling

Attempt to catch any exceptions and handle more gracefully.

mkhansen-intel commented on realsense_camera/src/base_nodelet.cpp:
```diff
@@ -45,8 +45,12 @@ namespace realsense_camera
+    }
 
     ROS_INFO_STREAM(nodelet_name_ << " - Stopping...");
     ros::shutdown();
```
Can you please add a check ros::isShuttingDown() here so that we don't call the shutdown again?

mdhorn committed:
Check status of shutdown before calling

kevincwells reviewed (approved):
These all seem like good checks, and I appreciate the new error messages.

mdhorn committed:
Display warning for hardcoded extrinsic

Display a warning message about the use of the hardcoded extrinsic
value for the color camera on ZR300 until librealsense supports it.
